MODULE.bazel

```diff
@@ -7,19 +7,122 @@ module(
 
 print("WARNING: The rules_rust Bazel module is still highly experimental and subject to change at any time. Only use it to try out bzlmod for now.")  # buildifier: disable=print
 
-bazel_dep(name = "platforms", version = "0.0.8")
-bazel_dep(name = "rules_cc", version = "0.0.9")
-bazel_dep(name = "bazel_skylib", version = "1.5.0")
+bazel_dep(
+    name = "bazel_skylib",
+    version = "1.5.0",
+)
+bazel_dep(
+    name = "platforms",
+    version = "0.0.8",
+)
+bazel_dep(
+    name = "rules_cc",
+    version = "0.0.9",
+)
+bazel_dep(
+    name = "rules_proto",
+    version = "5.3.0-21.7",
+)
 bazel_dep(
     name = "apple_support",
     version = "1.11.1",
     repo_name = "build_bazel_apple_support",
 )
+bazel_dep(
+    name = "protobuf",
+    version = "21.7",
+    repo_name = "com_google_protobuf",
+)
 
 internal_deps = use_extension("//rust/private:extensions.bzl", "internal_deps")
 use_repo(
     internal_deps,
+    "bazelci_rules",
```

Comment:
It's not super clear to me how much of this a real user would need to emulate and if there's any tooling we should be supplying to help... Is there an example somewhere?

Reply:
Not sure what you mean by emulate? The tooling is already there, if I understand your question correctly.

For example, I removed the bazelci_rules, then tried to build, and got the following:

```
$ bazel build --enable_bzlmod
Starting local Bazel server and connecting to it...
INFO: Invocation ID: 5c961a83-2149-45ba-a6b6-c0d403c9bfa9
DEBUG: /usr/local/google/home/msta/chromiumos_head/bazel_deps/rules_rust/MODULE.bazel:8:6: WARNING: The rules_rust Bazel module is still highly experimental and subject to change at any time. Only use it to try out bzlmod for now.
WARNING: /usr/local/google/home/msta/chromiumos_head/bazel_deps/rules_rust/MODULE.bazel:42:30: The module extension internal_deps defined in @rules_rust//rust/private:extensions.bzl reported incorrect imports of repositories via use_repo():

Not imported, but reported as direct dependencies by the extension (may cause the build to fail):
    bazelci_rules

 ** You can use the following buildozer command to fix these issues:

buildozer 'use_repo_add @rules_rust//rust/private:extensions.bzl internal_deps bazelci_rules' //MODULE.bazel:all
```

Reply:
What I mean is that things like `rules_rust_wasm_bindgen__anyhow-1.0.71` are internal dependencies of `rules_rust`, but are required if you want to use some parts of functionality in `rules_rust` from an external repo.

From what little I do know about bzlmod (again, apologies for my ignorance), one of the unergonomic pieces around it is that things like rules_go's third-party support requires you to `use_repo` every third-party library, even though they were initially loaded by a module extension via a `go.mod` file.

So my question is: Will users themselves need to `use_repo` the 21 `rules_rust_wasm_bindgen`-prefixed repos themselves if they want to use bindgen in their repo, or will this automatically be done because `rules_rust` does it, or is there some middle-ground where we can provide a utility function which will do it for them if they call it?

(None of this blocks merging the PR, which I will gladly do shortly, I'm just curious, and want to understand if there are things we should be doing to make things easier for our users to consume, or feedback we should be giving to the bzlmod folks about things which aren't ideal)

Reply:
Ah, I understand what you mean now. To clarify, this does two things:
* Building `//wasm_bindgen` with bzlmod should now work.
* Building `@rules_rust//wasm_bindgen` should now work

Longer explanation:
Users will just write `bazel_dep(name = "rules_rust", version = "...")` in their MODULE.bazel. This implicitly loads rules_rust's MODULE.bazel, which then declares `rules_rust_wasm_bindgen__anyhow-1.0.71`. However, since you didn't (and can't, because it's a private module extension) declare a dependency on it in your own MODULE.bazel, it's hidden from the main repo.

However, if I were to write `bazel build @rules_rust//wasm_bindgen`, it should work just fine, because it knows that `@rules_rust` can access it.